StopWatching() -> panic: send on closed channel

Seen on 7c8d46a1fca7ae5cebcb2cf98fd10966dae20dab after calling `Firebase.StopWatching()`

```
panic: send on closed channel

goroutine 124 [running]:
panic(0x53ece0, 0xc420a5a9b0)
  /usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
.../vendor/github.com/zabawaba99/firego.(*Firebase).Watch.func1(0xc4200e8fc0, 0xc420a5a309, 0xc4200e9020)
  /.../vendor/github.com/zabawaba99/firego/watch.go:99 +0x70
created by sigma/vendor/github.com/zabawaba99/firego.(*Firebase).Watch
  /.../vendor/github.com/zabawaba99/firego/watch.go:100 +0x132
```

--- a/watch.go
+++ b/watch.go
@@ -100,10 +100,7 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 	}()
 
 	go func() {
-		defer func() {
-			close(notifications)
-			close(stop)
-		}()
+		defer close(notifications)
 
 		for event := range events {
 			if closedManually {

--- a/watch_test.go
+++ b/watch_test.go
@@ -177,6 +177,41 @@ func TestWatchAuthRevoked(t *testing.T) {
 	assert.Equal(t, event.Data, `"token expired"`, "event data does not match")
 }
 
+func TestWatch_Issue66(t *testing.T) {
+	t.Parallel()
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		flusher, ok := w.(http.Flusher)
+		require.True(t, ok, "streaming unsupported")
+
+		w.Header().Set("Content-Type", "text/event-stream")
+		w.Header().Set("Cache-Control", "no-cache")
+		w.Header().Set("Connection", "keep-alive")
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		flusher.Flush()
+	}))
+	defer server.Close()
+
+	// create an initial sse connection
+	fb := New(server.URL, nil)
+	notifications := make(chan Event)
+	err := fb.Watch(notifications)
+	require.NoError(t, err)
+
+	// read the first connection
+	<-notifications
+
+	// close the server connection and read the error event
+	go server.Close()
+	<-notifications
+
+	// call stop watching - everything should be a-ok
+	fb.StopWatching()
+
+	_, ok := <-notifications
+	assert.False(t, ok)
+}
+
 func TestStopWatch(t *testing.T) {
 	t.Parallel()
 

